RUFAS/task_manager.py

```diff
@@ -555,6 +578,8 @@ def handle_post_processing(
             Whether to load data pool from file.
         export_input_data_to_csv: bool
             Whether to export the input data to a CSV file.
+        should_flush_im_pool: bool
```

Comment:
```suggestion
        should_flush_im_pool: bool, default True
```
For this arg, it might be better to not give it a default value and force every caller to explicitly say whether or not the pool should be flushed. It just feels big/important enough that making it as explicit as possible might be better than the alternative.